src/arch/x86_64.rs

```diff
@@ -72,10 +72,39 @@ pub extern "C" fn _start(info: &'static bootinfo::BootInfo) -> ! {
     mycelium_kernel::kernel_main(&bootinfo);
 }
 
-#[panic_handler]
-#[cfg(target_os = "none")]
-fn panic(info: &core::panic::PanicInfo) -> ! {
+pub(crate) fn oops(cause: &dyn core::fmt::Display) -> ! {
+    use core::fmt::Write;
+
+    unsafe { asm!("cli" :::: "volatile") }
     let mut vga = vga::writer();
```

Comment:
We should probably avoid trying to lock the VGA lock during the oops handler, which I think we do in methods like `set_color`, in case they are locked when the panic occurs.

It'd probably be reasonable (at least for now, as we don't have "threads") to attempt to take the lock and unsafely bypass it if we can't.

Reply:
hmm, yeah...i didn't initially want to add this, but i think it makes sense for our mutex to have an `unsafe fn force_unlock` or something specifically for cases where we _know_ a previously acquired lock will never be used again (basically when we're oopsing). even when we are running on multiple cores, i think we would want to kill them pretty early in the oops process...

Reply:
Yeah. I imagine we want to halt all cores before interacting with VGA at all, at which point forcing unlocks should be "safe".